recipes/perl/assets/create_assetGroup.pl

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env perl
+
+use LWP::UserAgent;
+use LWP::Protocol::https;
+print "LWP::UserAgent: ".LWP::UserAgent->VERSION,"\n";
+print "LWP::Protocol::https: ".LWP::Protocol::https->VERSION,"\n";
+use JSON;
+use Getopt::Long qw(GetOptions);
+
+
+#
+# The token is the key to the NetBackup AuthN/AuthZ scheme.  You must login and get a token
+# and use this token in your Authorization header for all subsequent requests.  Token validity
+# is fixed at 24 hours
+#
+
+my $content_type = "application/vnd.netbackup+json; version=4.0";
+my $protocol = "https";
+my $port = "1556";
+my $nbmaster;
+my $username;
+my $password;
+my $domainName;
+my $domainType;
+my $token;
+my $base_url;
+
+my $ua = LWP::UserAgent->new(
+                ssl_opts => { verify_hostname => 0, verify_peer => 0},
+            );
+
+# subroutines for printing usage and library information required to run the script.
+sub print_usage {
+    print("\n\nUsage:");
+    print("\nperl get_vmware_assets -nbmaster <masterServer> -username <username> -password <password> [-domainName <domainName>] [-domainType <domainType>] \n\n\n");
+}
+
+sub print_disclaimer {
+    print("--------------------------------------------------------\n");
+    print("--       This script requires Perl 5.20.2 or later    --\n");
+    print("--------------------------------------------------------\n");
+    print("Executing this library requires some additional libraries like \n\t'LWP' \n\t'JSON'\ \n\t'Getopt'\ \n\n");
+    print("You can specify the 'nbmaster', 'username', 'password', 'domainName', 'domainType' as command-line parameters\n");
+    print_usage();
+}
+
+
+# subroutine to process user input
+sub user_input {
+    GetOptions(
+        'nbmaster=s' => \$nbmaster,
+        'username=s' => \$username,
+        'password=s' => \$password,
+        'domainName=s' => \$domainName,
+        'domainType=s' => \$domainType,
+    ) or die print_usage();
+
+    if ($nbmaster eq "") {
+        print("Please provide the value for 'nbmaster'");
+        exit;
+    }
+
+    if ($username eq "") {
+        print("Please provide the value for 'username'");
+        exit;
+    }
+
+    if ($password eq "") {
+        print("Please provide the value for 'password'");
+        exit;
+    }
+
+    $base_url = "$protocol://$nbmaster:$port/netbackup";
+}
+
+sub login {
+
+    my @argument_list = @_;
+    $base_url = $argument_list[0];
+    my $username = $argument_list[1];
+    my $password = $argument_list[2];
+    my $domainName = $argument_list[3];
+    my $domainType = $argument_list[4];
+
+    my $url = "$base_url/login";
+
+    my $req = HTTP::Request->new(POST => $url);
+    $req->header('content-type' => 'application/json');
+
+    if ($domainName eq "" && $domainType eq "") {
+        $post_data = qq({ "userName": "$username", "password": "$password" });
```

Comment:
This will be a JSON syntax error if `$username` or `$password` contains characters special to JSON. I'd recommend building a Perl hash and then converting it with `encode_json`.

Reply:
my password already contains special characters.. why am not seeing that syntax error.

Reply:
Quotation marks and backslashes are what would be relevant in this context.